src/controllers/auth.controller.js

```diff
@@ -0,0 +1,170 @@
+const userService = require('../services/user.service');
+const jwtService = require('../services/jwt.service');
+const tokenService = require('../services/token.service');
+const bcrypt = require('bcrypt');
+const { validateEmail,
+  validatePassword,
+  validateName } = require('../utils/validators');
+
+async function generateTokens(res, user) {
+  const normalizedUser = userService.normalize(user);
+  const accessToken = jwtService.sign(normalizedUser);
+  const refreshAccessToken = jwtService.signRefresh(normalizedUser);
+
+  await tokenService.save(normalizedUser.id, refreshAccessToken);
+
+  res.cookie('refreshToken', refreshAccessToken, {
+    HttpOnly: true,
+    maxAge: 30 * 24 * 60 * 60 * 1000,
+  });
+
+  res.send({
+    user: normalizedUser,
+    accessToken,
+  });
+}
+
+const register = async(req, res, next) => {
+  const { name, email, password } = req.body;
+
+  const errors = {
+    email: validateEmail(email),
+    password: validatePassword(password),
+    name: validateName(name),
+  };
+
+  if (errors.email || errors.password || errors.name) {
+    throw ApiError.badRequest('Bad request', errors);
+  }
+
+  await userService.register(name, email, password);
+  res.send({ message: 'OK' });
+};
+
+const activate = async(req, res) => {
+  const { activationToken } = req.params;
+  const user = await User.findOne({ where: { activationToken } });
+
+  if (!user) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  user.activationToken = null;
+  user.save();
+
+  res.send(user);
+};
+
+const login = async(req, res) => {
+  const { email, password } = req.body;
+  const user = await userService.findByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('No such user');
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.badRequest('Wrong password');
+  };
+
+  await generateTokens(res, user);
+};
+
+const refresh = async(req, res) => {
+  const { refreshToken } = req.cookies;
+  const userData = await jwtService.verifyRefresh(refreshToken);
+  const token = await tokenService.getByToken(refreshToken);
+
+  if (!userData || !token) {
+    throw ApiError.unauthorized();
+  };
+
+  const user = await userService.findByEmail(userData.email);
+
+  await generateTokens(res, user);
+};
+
+const logout = async(req, res) => {
+  const { refreshToken } = req.cookies;
+  const userData = await jwtService.verifyRefresh(refreshToken);
+
+  if (!userData || !refreshToken) {
+    throw ApiError.unauthorized();
+  }
+
+  await tokenService.remove(userData.id);
+
+  res.sendStatus(204);
+};
+
+const restorePassword = async(req, res) => {
+  const { email } = req.body;
+
+  if (!email) {
+    throw ApiError.notFound('Email not found');
+  }
+
+  await userService.restorePassword(email);
+
+  res.sendStatus({ message: 'Restore password is sent!' });
+};
+
+const useRestore = async(req, res) => {
+  const { restoreCode } = req.params;
```

Comment:
The same issue here. Look through your code and move a work (I mean validation work or check the existence of needed fields in request object to middlewares)